### PR TITLE
DOC: Correct inconsistent description on default DateOffset setting

### LINKED ISSUE #36512: DOC: default value for pd.DateOffset() is 24 hours instead of 1 calendar day
```python
>>>pd.__version__
1.1.1
>>>ts = pd.Timestamp('2011-3-27 00:00:00', tz='Europe/Helsinki')
>>>ts + pd.DateOffset(days=1)
Timestamp('2011-03-28 00:00:00+0300', tz='Europe/Helsinki')
>>>ts + pd.DateOffset()
Timestamp('2011-03-28 01:00:00+0300', tz='Europe/Helsinki')
>>>ts + pd.DateOffset(hours=24)
Timestamp('2011-03-28 01:00:00+0300', tz='Europe/Helsinki')
```

The default action for pd.DateOffset seems to be equal to pd.DateOffset(hours=24) instead of pd.DateOffset(days=1).
But the user guide says DateOffset object defaults to 1 calendar day.

捕获

### PR DESCRIPTION
**Pull request by GYHHAHA**
correct inconsistent doc description on default DateOffset setting

- [x] closes #36512

### CODE REVIEW

File: doc/source/user_guide/timeseries.rst [hunk 1]
```diff
@@ -831,6 +831,8 @@ arithmetic operator (``+``) or the ``apply`` method can be used to perform the s
    ts + pd.Timedelta(days=1)
    # Respects calendar time
    ts + pd.DateOffset(days=1)
+   # Default DateOffset is 24 hours instead of a calendar day since Hour is one of subclasses for DateOffset
```
Comment (arw2019) [inline doc/source/user_guide/timeseries.rst:834]: Personally I'd leave this out as we already document the default behavior below

**Review by arw2019 — APPROVED** [review]
thanks @GYHHAHA

**Comment by arw2019** [comment]
Also heads up there are linting issues
```
##[error]doc/source/user_guide/timeseries.rst:834:83:E501:line too long (105 > 79 characters)
```
to be resolved before merge

**Comment by GYHHAHA** [comment]
why fail and how can I fix?

**Comment by MarcoGorelli** [comment]
What's wrong with "one calendar day"?

**Comment by GYHHAHA** [comment]
```
>>>ts = pd.Timestamp('2011-3-27 00:00:00', tz='Europe/Helsinki')
```
One calendar day will get the result below.
```
>>>ts + pd.DateOffset(days=1)
Timestamp('2011-03-28 00:00:00+0300', tz='Europe/Helsinki')
```
But now the default action is add absolute 24 hours.
```
>>>ts + pd.DateOffset()
Timestamp('2011-03-28 01:00:00+0300', tz='Europe/Helsinki')
>>>ts + pd.DateOffset(hours=24)
Timestamp('2011-03-28 01:00:00+0300', tz='Europe/Helsinki')
```

**Comment by MarcoGorelli** [comment]
Ah, thanks!

**Comment by GYHHAHA** [comment]
So how can I fix these failings? Seems that I only change one line.

**Comment by MarcoGorelli** [comment]
They're unrelated, you'll probably need to merge upstream/master later

File: doc/source/user_guide/timeseries.rst [hunk 2]
```diff
@@ -846,7 +846,7 @@ into ``freq`` keyword arguments. The available date offsets and associated frequ
     :header: "Date Offset", "Frequency String", "Description"
     :widths: 15, 15, 65
 
-    :class:`~pandas.tseries.offsets.DateOffset`, None, "Generic offset class, defaults to 1 calendar day"
+    :class:`~pandas.tseries.offsets.DateOffset`, None, "Generic offset class, defaults to 24 hours"
```
Comment (jreback) [inline doc/source/user_guide/timeseries.rst:849]: cc @mroeschke

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by GYHHAHA** [comment]
But still a little strange about the default value, why not just set it a calendar day since the offset objects are specially designed for calendar duration rules?

**Comment by mroeschke** [comment]
I think this should ideally default to calendar day instead 24 hours as well. This would require a deprecation cycle related to https://github.com/pandas-dev/pandas/issues/7418 that would deprecate all arguments that don't end with `s`.

Would you be interested in making that change instead @GYHHAHA?

**Comment by GYHHAHA** [comment]
I will take a try.

File: pandas/_libs/tslibs/offsets.pyx [hunk 3]
```diff
@@ -319,7 +319,8 @@ cdef _determine_offset(kwds):
             # sub-daily offset - use timedelta (tz-aware)
             offset = timedelta(**kwds_no_nanos)
     else:
-        offset = timedelta(1)
+        offset = relativedelta(**{'days':1})
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:322]: @mroeschke I don't know how we can easily deprecate this, nor can we simply change this.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by GYHHAHA** [comment]
For # 7418, can we just convert any argument in `kwds_no_nanos` that end with s to the one with s? @jreback

**Comment by GYHHAHA** [comment]
Also since the `_determine_offset` doesn't handle the nanosecond or nanoseconds, thus here is a bug related to this.
```
>>>ts = pd.Timestamp('2020-1-1')
>>>ts + pd.DateOffset(nanoseconds=1000)
Timestamp('2020-01-02 00:00:00')
```
Caused by the default `offset = timedelta(1)` setting.

File: pandas/_libs/tslibs/offsets.pyx (hunk 3 above)
Comment (mroeschke) [reply pandas/_libs/tslibs/offsets.pyx:322]: Can't simply change this in the next release, but I think we could deprecate the keyword arguments and slate for removal in 2.0.

**Comment by GYHHAHA** [comment]
Ok, I will keep this in original form, and should I continue to change the "1 calendar day" to "24 hours" in the DOC?
1) Maybe a warning need to be added when use singular arguments.
2) I find in # 7418 `ts.replace(pd.DateOffset(...))` is suggested, is this a good way to make these replacement actions?
3) The above-mentioned nanoseconds BUG needs a fix, but since this relates to the default setting, will it be troublesome to handle?

Thanks. @jreback @mroeschke

**Comment by mroeschke** [comment]
Yes I think for this PR, you can just make the documentation change.

You can follow up with another PR to deprecate the DateOffset arguments that do not end in  `s` if interested.

Also if you could submit a bug report with the nanosecond example, that'd be great.

**Comment by GYHHAHA** [comment]
Here are also some problems.

DOC says "However, all DateOffset subclasses that are an hour or smaller (Hour, Minute, Second, Milli, Micro, Nano) behave like Timedelta and respect absolute time.".

But though `DateOffset(days=1)` indeed use a calendar day, still `offsets.Day(1)` use 24 absolute hours.

Is it necessary to keep this consistency?

Besides, in #36592 I suggest to add `replace` method for time-related replacement, but now I also think it may not be a good choice.

However, we can only use `ts + pd.DateOffset(day=5)` to change the absolute `year/day/hour/... ` use the singular form argument which will be deprecated (if we make this deprecation), thus how can we do this absolute time replacement by other offsets object? (like changing year to 2000 or day to 22 or month to 5)

Thanks! @mroeschke

**Comment by mroeschke** [comment]
See https://github.com/pandas-dev/pandas/issues/22864 about the `offsets.Day` consistency.

For absolute time changes, I think users should just stick to using `Timedelta` objects. `DateOffset` object should only operate with relative time changes.

**Comment by GYHHAHA** [comment]
agree with jbrockmendel

> If we're talking reasonably long-term, I think we should get rid of the Tick classes entirely and use Timedelta in their place

It maybe will clear to only consider the relative time (day, week, month ,year, ...) for offsets and tick time (nano, ..., minute, hour, absolute day) for timedelta.

**Comment by jreback** [comment]
@mroeschke this lgtu?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @GYHHAHA

**Pull request merged**